cli/azd/cmd/up.go

```diff
@@ -130,6 +130,14 @@ func (u *upAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 		u.console.Message(ctx, output.WithGrayFormat("Note: Running custom 'up' workflow from azure.yaml"))
 	}
 
+	envValue := u.flags.EnvFlag.EnvironmentName
+	for _, step := range upWorkflow.Steps {
+		if envValue != "" {
+			stepArgs := append(step.AzdCommand.Args, "-e", envValue)
+			step.AzdCommand.Args = stepArgs
+		}
+	}
```

Comment:
@vhvb1989 Thanks for your feedback!

When debugging the `azd up -e <environment>` command, we found that the actual executed steps (such as `azd package`, `azd provision`, and `azd deploy`) did not include the `-e` flag, which caused these commands to use the default environment instead of the environment specified by the user through `-e`. To ensure that these subcommands can correctly use the environment specified by `-e`, I manually set it in the code.

If you think there is a better solution, you are welcome to discuss further in the issue https://github.com/Azure/azure-dev/issues/4411#issuecomment-2482358893!

Reply:
The current solution wouldn't work if the user added the workflow step to reference an azd command not accepting `-e`. An immediate solution doesn't come to mind, but will require some additional team discussion.